src/bun.js/webcore/blob.zig

```diff
@@ -1424,22 +1426,20 @@ pub const Blob = struct {
         return blob_;
     }
 
-    fn estimatedByteSize(this: *Blob) usize {
+    fn calculateEstimatedByteSize(this: *Blob) usize {
         // in-memory size. not the size on disk.
-        if (this.size != Blob.max_size) {
-            return this.size;
-        }
-
-        const store = this.store orelse return 0;
-        if (store.data == .bytes) {
-            return store.data.bytes.len;
-        }
-
-        return 0;
-    }
-
-    pub fn estimatedSize(this: *Blob) callconv(.C) usize {
-        var size = this.estimatedByteSize() + @sizeOf(Blob);
+        var size: usize = @sizeOf(Blob);
+
+        size += if (this.size != Blob.max_size)
+            @intCast(this.size)
```

Comment:
We should not report the size of `.file` blobs as the real memory size

This value should only be respected when `this.store` is  .bytes